File: java/base/src/main/java/vsdk/toolkit/common/dataStructures/KDQuery.java
package vsdk.toolkit.common.dataStructures;

public class KDQuery {
    public float[] point;
    public int wantedN;
    public int foundN;
    public boolean notFilled;
    public Object[] results;
    public float[] distances;
    public float maximumDistance;
    public float sqrRadius;
    public short excludeFlags;

    public KDQuery() {
    }

    public void print() {
        System.out.printf("Point X %g, Y %g, Z %g%n", point[0], point[1], point[2]);
        System.out.printf("Wanted N: %d, found N: %d%n", wantedN, foundN);
        System.out.printf("maximumDistance %g%n", maximumDistance);
        System.out.printf("sqrRadius %g%n", sqrRadius);
        System.out.printf("excludeFlags %x%n", (int)excludeFlags);
    }
}

File: cpp/base/src/main/vsdk/toolkit/common/dataStructures/KDQuery.cpp
#include "java/lang/System.h"
#include "vsdk/toolkit/common/dataStructures/KDQuery.h"

KDQuery::KDQuery():
    point(),
    wantedN(),
    foundN(),
    notFilled(),
    results(),
    distances(),
    maximumDistance(),
    sqrRadius(),
    excludeFlags()
{
}

void
KDQuery::print() const {
    java::System::out.printf("Point X %g, Y %g, Z %g\n", point[0], point[1], point[2]);
    java::System::out.printf("Wanted N: %i, found N: %i\n", wantedN, foundN);
    java::System::out.printf("maximumDistance %g\n", maximumDistance);
    java::System::out.printf("sqrRadius %g\n", sqrRadius);
    java::System::out.printf("excludeFlags %x\n", static_cast<int>(excludeFlags));
}
